hedvig-market/src/main/kotlin/com/hedvig/android/market/MarketManager.kt

```diff
@@ -1,51 +1,56 @@
 package com.hedvig.android.market
 
-import android.annotation.SuppressLint
-import android.content.Context
-import androidx.preference.PreferenceManager
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.runBlocking
 
 interface MarketManager {
   val enabledMarkets: List<Market>
-  var market: Market?
+  val market: Market?
+  suspend fun setMarket(market: Market)
+  suspend fun removeMarket()
+  fun observeMarket(): Flow<Market?>
 }
 
 internal class MarketManagerImpl(
-  context: Context,
   isDebug: Boolean,
+  private val dataStore: DataStore<Preferences>,
 ) : MarketManager {
 
-  private val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
-
   override val enabledMarkets = listOfNotNull(
     Market.SE,
     Market.NO,
     Market.DK,
     if (isDebug) Market.FR else null,
   )
+  override val market: Market?
+    get() = runBlocking { observeMarket().first() }
```

Comment:
Yeah we were reading this in a blocking manner anyway before, I guess we're keeping it to avoid having to update *all* call sites. Just gonna have to be a bit careful if this is blocking for too long cause it's gonna happen on the UI thread. But we can let if be for now, since I am hoping it won't be too long anyway.